### PR TITLE
Initial Patch for RISC-V support.

### PR DESCRIPTION
**Pull request by viewizard**
Based on CoreCLR changes.

CC @clamp03 @wscho77 @HJLeee @JongHeonChoi @t-mustafin @gbalykov

### CODE REVIEW

**Comment by viewizard** [comment]
@microsoft-github-policy-service agree company="Samsung"

**Comment by leculver** [comment]
I will accept this pull request with the understanding that I can't test this or keep it working if something breaks or changes.

RISC-V support is very welcome, but if it's broken in the future and I have no way to debug it, then I may strip support out if no one volunteers to fix it...because I may not have a way to support it myself.  ( After pinging you all and requesting help of course, I'm not saying I'll strip it out day 1.  :) )

File: src/Microsoft.Diagnostics.Runtime/DataReaders/Registers/RiscV64Context.cs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+        [Register(RegisterType.General)]
+        [FieldOffset(0x38)]
+        public ulong T1;
+
+        [Register(RegisterType.General)]
+        [FieldOffset(0x40)]
+        public ulong T2;
+
+        [Register(RegisterType.General)]
```
Comment (gbalykov) [inline src/Microsoft.Diagnostics.Runtime/DataReaders/Registers/RiscV64Context.cs:61]: `[Register(RegisterType.General | RegisterType.FramePointer)]`

File: src/Microsoft.Diagnostics.Runtime/DataReaders/Registers/RiscV64Context.cs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        [FieldOffset(0xf8)]
+        public ulong T5;
+
+        [Register(RegisterType.General)]
+        [FieldOffset(0x100)]
+        public ulong T6;
+
+        [Register(RegisterType.General)]
```
Comment (gbalykov) [inline src/Microsoft.Diagnostics.Runtime/DataReaders/Registers/RiscV64Context.cs:157]: It seems that `#region Control Registers`  is needed before this

Comment (gbalykov) [inline src/Microsoft.Diagnostics.Runtime/DataReaders/Registers/RiscV64Context.cs:157]: `[Register(RegisterType.Control | RegisterType.ProgramCounter)]`

File: src/Microsoft.Diagnostics.Runtime/DataReaders/Registers/RiscV64Context.cs [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+namespace Microsoft.Diagnostics.Runtime
+{
+    /// <summary>
+    /// RISC-V-specific thread context.
+    /// </summary>
+    [StructLayout(LayoutKind.Explicit, Pack = 1)]
+    public struct RiscV64Context
+    {
+        public const uint Context = 0x01000000;
```
Comment (gbalykov) [inline src/Microsoft.Diagnostics.Runtime/DataReaders/Registers/RiscV64Context.cs:14]: What does this value mean?

**Review by gbalykov — COMMENTED** [review]
Have few notes, please fix in next PR

**Pull request merged**